### PR TITLE
Add don't bin option for temporal bucketing

### LINKED ISSUE #19684: It's not possible to group by a temporal column WITHOUT using bucketing
I have data where the timestamps are a few distinct values, and do NOT want to apply temporal bucketing to them. However it's impossible to "Summarize" by a timestamp column without choosing temporal bucketing.

image

### LINKED ISSUE #35550: "Don't bin" is not selected on the picker in temporal bucketing
**Context**
Even if we select `Don't bin` and then open menu again, `Don't bin` option is not highlighted as selected

- issue links: https://github.com/metabase/metabase/issues/19684 and https://github.com/metabase/metabase/issues/35546

https://github.com/metabase/metabase/assets/125459446/d27e815a-3252-4add-8fb2-1fb8f918bf5a

### PR DESCRIPTION
**Pull request by uladzimirdev**
Closes https://github.com/metabase/metabase/issues/19684
Closes https://github.com/metabase/metabase/issues/35550

### Description

Add `Don't bin` option for temporal bucketing.

Doesn't cover https://github.com/metabase/metabase/issues/35546

### How to verify

Create a new question like in [figma](https://www.figma.com/file/HPg8IDPGo9seZafglWmhIk/Querying-Bug-Fixes?type=design&node-id=101-30315&mode=design&t=zAUnLpMXJI49RCXz-0)
`Don't bin` should be on the list

### Demo

https://github.com/metabase/metabase/assets/125459446/e67dbc3d-5765-46a0-8f8f-1379dd2d7157

### Checklist

- [x] Tests have been added/updated to cover changes in this PR

### CODE REVIEW

File: frontend/src/metabase/query_builder/components/notebook/steps/BreakoutStep/BreakoutStep.unit.spec.tsx [hunk 1]
```diff
@@ -35,14 +35,14 @@ function createQueryWithBinning(bucketName = "10 bins") {
   return { query, columnInfo: Lib.displayInfo(query, 0, columnWithBinning) };
 }
 
-function createQueryWithTemporalBreakout() {
+function createQueryWithTemporalBreakout(bucketName = "Quarter") {
```
Comment (kamilmielnik) [inline frontend/src/metabase/query_builder/components/notebook/steps/BreakoutStep/BreakoutStep.unit.spec.tsx:38]: IMO `bucketName` should not have default value

**Review by kamilmielnik — APPROVED** [review]
Looks good, works well :+1:

File: frontend/src/metabase/query_builder/components/notebook/steps/BreakoutStep/BreakoutStep.unit.spec.tsx (hunk 1 above)
Comment (uladzimirdev) [reply frontend/src/metabase/query_builder/components/notebook/steps/BreakoutStep/BreakoutStep.unit.spec.tsx]: removed

File: frontend/src/metabase/common/components/QueryColumnPicker/BucketPickerPopover/BucketPickerPopover.unit.spec.tsx [hunk 2]
```diff
@@ -62,7 +62,8 @@ describe("BucketPickerPopover", () => {
 
     userEvent.click(screen.getByRole("button", { name: "More…" }));
 
-    expect(screen.getAllByRole("menuitem")).toHaveLength(buckets.length);
+    const bucketsSize = buckets.length + 1; // includes Don't bin
```
[the remark this answers is not in the document]
Comment (uladzimirdev) [reply frontend/src/metabase/common/components/QueryColumnPicker/BucketPickerPopover/BucketPickerPopover.unit.spec.tsx]: fixed, nice tip!

File: frontend/src/metabase/common/components/QueryColumnPicker/BucketPickerPopover/TemporalBucketPickerPopover.tsx [hunk 3]
```diff
@@ -8,22 +8,22 @@ import {
...
   onSelect,
   ...props
 }: CommonBucketPickerProps) {
   const selectedBucket = useMemo(() => Lib.temporalBucket(column), [column]);
 
   const items = useMemo(
-    () => buckets.map(bucket => getBucketListItem(query, stageIndex, bucket)),
+    () => [
+      ...buckets.map(bucket => getBucketListItem(query, stageIndex, bucket)),
+      { displayName: t`Don't bin`, bucket: null },
```
Comment (ranquild) [inline frontend/src/metabase/common/components/QueryColumnPicker/BucketPickerPopover/TemporalBucketPickerPopover.tsx:25]: I think we could set `selected: !selectedBucket` here and avoid the check in `checkBucketIsSelected`

**Review by ranquild — COMMENTED** [review]
Minor comment

File: frontend/src/metabase/common/components/QueryColumnPicker/BucketPickerPopover/TemporalBucketPickerPopover.tsx (hunk 3 above)
Comment (uladzimirdev) [reply frontend/src/metabase/common/components/QueryColumnPicker/BucketPickerPopover/TemporalBucketPickerPopover.tsx]: fixed. I'm not 100% sure we need memoization here, but if `getBucketListItem` is slow, maybe it makes sense.

check on isEditing is required for the initial rendering (Don't bin is preselected even if we didn't add Created At to the breakout)

**Review by ranquild — APPROVED** [review]

**Pull request merged**